sequencer/src/state.rs

```diff
@@ -407,22 +402,22 @@ impl<'a> StateCatchup for SqlStateCatchup<'a> {
 
 pub async fn update_state_storage(
     storage: &mut SqlDataSource<SeqTypes, Provider>,
-    instance: NodeState,
-    parent_leaf: LeafQueryData<SeqTypes>,
-    proposed_leaf: LeafQueryData<SeqTypes>,
+    instance: &NodeState,
+    parent_leaf: &LeafQueryData<SeqTypes>,
+    proposed_leaf: &LeafQueryData<SeqTypes>,
 ) -> anyhow::Result<()> {
     let proposed_leaf = proposed_leaf.leaf();
     let parent_leaf = parent_leaf.leaf();
-    let header = proposed_leaf.get_block_header();
+    let header = parent_leaf.get_block_header();
```

Comment:
Ah, but `header` is still the new header we're applying, so it has to come from the `proposed_leaf`. In other words `header` to apply and `validated_state` come from different leaves